___Execucao_com 2 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 3.21628
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:03
	Energy:	0.0001 kWh
	CO2eq:	0.0667 g
	This is equivalent to:
	0.0006 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:03
	Energy:	0.0001 kWh
	CO2eq:	0.0667 g
	This is equivalent to:
	0.0006 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 2 Treads_concluida___
___Execucao_com 4 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 1.77241
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:02
	Energy:	0.0001 kWh
	CO2eq:	0.0410 g
	This is equivalent to:
	0.0004 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:02
	Energy:	0.0001 kWh
	CO2eq:	0.0410 g
	This is equivalent to:
	0.0004 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 4 Treads_concluida___
___Execucao_com 6 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 1.22850
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0312 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0312 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 6 Treads_concluida___
___Execucao_com 8 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 1.02749
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0273 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0273 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 8 Treads_concluida___
___Execucao_com 10 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.89463
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0225 g
	This is equivalent to:
	0.0002 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 10 Treads_concluida___
___Execucao_com 12 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.73339
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 12 Treads_concluida___
___Execucao_com 14 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.60686
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 14 Treads_concluida___
___Execucao_com 16 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.55519
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 16 Treads_concluida___
___Execucao_com 18 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.49464
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0099 g
	This is equivalent to:
	0.0001 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 18 Treads_concluida___
___Execucao_com 20 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.51544
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 20 Treads_concluida___
___Execucao_com 22 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.51127
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 22 Treads_concluida___
___Execucao_com 24 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.66894
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 24 Treads_concluida___
___Execucao_com 26 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.64095
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 26 Treads_concluida___
___Execucao_com 28 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.58882
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 28 Treads_concluida___
___Execucao_com 30 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.58429
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 30 Treads_concluida___
___Execucao_com 32 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.53302
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 32 Treads_concluida___
___Execucao_com 34 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.47071
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 34 Treads_concluida___
___Execucao_com 36 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.44779
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 36 Treads_concluida___
___Execucao_com 38 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.45520
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 38 Treads_concluida___
___Execucao_com 40 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41022
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 40 Treads_concluida___
___Execucao_com 42 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42055
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 42 Treads_concluida___
___Execucao_com 44 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41957
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 44 Treads_concluida___
___Execucao_com 46 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42572
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 46 Treads_concluida___
___Execucao_com 48 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41540
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 48 Treads_concluida___
___Execucao_com 50 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42178
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 50 Treads_concluida___
___Execucao_com 52 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.49730
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 52 Treads_concluida___
___Execucao_com 54 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.63286
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 54 Treads_concluida___
___Execucao_com 56 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42637
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 56 Treads_concluida___
___Execucao_com 58 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.40296
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 58 Treads_concluida___
___Execucao_com 60 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.40440
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 60 Treads_concluida___
___Execucao_com 62 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41606
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 62 Treads_concluida___
___Execucao_com 64 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42129
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 64 Treads_concluida___
___Execucao_com 66 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.40151
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 66 Treads_concluida___
___Execucao_com 68 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.34744
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 68 Treads_concluida___
___Execucao_com 70 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.40588
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 70 Treads_concluida___
___Execucao_com 72 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.39210
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 72 Treads_concluida___
___Execucao_com 74 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.43470
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 74 Treads_concluida___
___Execucao_com 76 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42993
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 76 Treads_concluida___
___Execucao_com 78 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41026
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 78 Treads_concluida___
___Execucao_com 80 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.50231
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 80 Treads_concluida___
___Execucao_com 82 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.45870
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 82 Treads_concluida___
___Execucao_com 84 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.50565
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 84 Treads_concluida___
___Execucao_com 86 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.49294
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 86 Treads_concluida___
___Execucao_com 88 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.73536
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 88 Treads_concluida___
Alerta: Menos de 10 log_bruto
